### PR TITLE
Use mpi-smp charm on CI

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

To be able to test that charm messages are working properly, we have to be able to run on multiple charm-nodes. This requires us to have a non-multicore build of charm on CI. I chose MPI because that's what we often use in our production environments.

To keep the size of the container down, I replaced the clang-multicore build of charm with a clang-mpi-smp build.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by knelli2** [comment]
One of the CI builds is going to fail until the container is updated to have charm built with MPI.

**Review by wthrowe — DISMISSED** [review]

File: .github/workflows/Tests.yaml
Comment (nilsvu) [inline .github/workflows/Tests.yaml]: Use the env var `${CHARM_ROOT}` from the container here and in the other places. You can actually delete this line and the build will use `${CHARM_ROOT}` automatically.

File: .github/workflows/Tests.yaml
Comment (nilsvu) [inline .github/workflows/Tests.yaml]: Delete this line, and also `charm_name` above. I believe this entry in the matrix is only needed if you want to _add_ builds to the matrix, not to modify an existing one. Instead, just set `CHARM_ROOT` for one of the builds and fall back to the default, like `BUILD_PYTHON_BINDINGS` below.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: .github/workflows/Tests.yaml
Comment (nilsvu) [reply .github/workflows/Tests.yaml]: Also, if you move this commit into a separate PR that we can merge first, people can already rebase on this and fewer PRs will get disrupted by the new Docker container.

Comment (knelli2) [reply .github/workflows/Tests.yaml]: See #4677.

**Comment by knelli2** [comment]
Squashed in changes. Now dependent on #4677.

**Comment by knelli2** [comment]
Rebased on develop. No longer dependent on #4677

**Review by nilsvu — DISMISSED** [review]
I can build and push the container on Monday, if no one has done it by then.

**Comment by nilsdeppe** [comment]
LGTM. Just need to coordinate updating the container since it might cause existing PRs to fail CI (which is fine!)

**Comment by nilsvu** [comment]
@knelli2 I pushed the new container and also added a commit that upgrades Blaze in the container. Please pull that commit to your local branch before force-pushing again (only in case you need to force push, we might just merge this PR as-is if CI passes).

**Comment by knelli2** [comment]
Ok I'll wait until CI finishes to do anything. I'll check back later and if things are failing or something went wrong, I'll rebase on latest develop and force push

**Comment by nilsvu** [comment]
Just to clarify: my commit that adds Blaze is _on this branch_, not on develop. So just `git pull` before rebasing / pushing.

**Comment by knelli2** [comment]
Ahh yes ok, thanks for clarifying. I totally would've forgotten to do that lol

**Comment by nilsvu** [comment]
@knelli2 the new Charm build wasn't found by CMake. Maybe the directory name just isn't correct?

**Comment by nilsvu** [comment]
We encountered some issues with Blaze 3.8.2, so I pushed the container without the Blaze upgrade and restarted CI on this PR

**Review by nilsvu — APPROVED** [review]

**Pull request merged**
